File: Regression/RelaxationTerms/DecisionTree/sw_air_code/opt1/in_con.m
function [Y] = in_con
% функция расчета начальных параметров смеси O2/N2 на УВ
% эксперимент wurster (1991)

global k m v0 T0 xc; 
% n0, v0, T0 -- параметры смеси перед УВ
% xc -- состав смеси

% искомые параметры за фронтом УВ
syms n v t

C1 = v0^2*sum(xc.*m)/(k*T0);
C2 = 0.5*C1;

S = [n*v == 1,...
     n*t+n*v^2*C1 == 1+C1,...
     3.5*sum(xc(1:3))*t+2.5*sum(xc(4:5))*t+C2*v^2 == ...
     3.5*sum(xc(1:3))+2.5*sum(xc(4:5))+C2];

N = vpasolve(S,[n,v,t],[10,0.1,100]);
X = [N.n;N.v;N.t];
sol = double(X);

% выбор нетривиального решения
sol1 = sol([1 3 5]);
sol2 = sol([2 4 6]);
if any(sol1-1)
    Y = sol1;
elseif any(sol2-1)
    Y = sol2;
end

File: matlab2julia/bi_mix_sts_julianized/in_con.jl
function in_con()

n = Sym("n")
v = Sym("v")
t = Sym("t")
n, v, t = symbols("n, v, t", real=true)

xm = xc[1]
xa = xc[2]

C1 = sum(m.*xc)*v0^2/(k*T0)
C2 = 0.5*C1

S = [n*v ⩵ 1, n*t+n*v^2*C1 ⩵ (1+C1), 3.5*xm*t+2.5*xa*t+v^2*C2 ⩵ (3.5*xm+2.5*xa+C2)]
N = SymPy.solve(S, n, v, t)

sol1 = collect(values(N[2]))
sol2 = collect(values(N[1]))

#FIXME
#if any(sol1 != 0)
#  Y = sol1
#elseif any(sol2 != 0)
#  Y = sol2
#end

return sol1

end
